src/python/grpcio_tests/tests/unit/_invalid_metadata_test.py

```diff
@@ -0,0 +1,169 @@
+    expected_error_details = "metadata was invalid: %s" % metadata
+    with self.assertRaises(ValueError) as exception_ctx:
+      response, call = self._unary_unary.with_call(request, metadata=metadata)
+    self.assertEqual(expected_error_details, exception_ctx.exception.message)
+
+  def testUnaryRequestFutureUnaryResponse(self):
+    request = b'\x07\x08'
+    metadata = (('InVaLiD', 'UnaryRequestFutureUnaryResponse'),)
+    expected_error_details = "metadata was invalid: %s" % metadata
+    response_future = self._unary_unary.future(request, metadata=metadata)
+    with self.assertRaises(grpc.RpcError) as exception_ctx:
+      response = response_future.result()
+    self.assertEqual(exception_ctx.exception.details(), expected_error_details)
+    self.assertEqual(exception_ctx.exception.code(), grpc.StatusCode.INTERNAL)
+    self.assertEqual(response_future.details(), expected_error_details)
+    self.assertEqual(response_future.code(), grpc.StatusCode.INTERNAL)
+
+  def testUnaryRequestStreamResponse(self):
+    request = b'\x37\x58'
+    metadata = (('InVaLiD', 'UnaryRequestStreamResponse'),)
+    expected_error_details = "metadata was invalid: %s" % metadata
+    response_iterator = self._unary_stream(request, metadata=metadata)
+    with self.assertRaises(grpc.RpcError) as exception_ctx:
+      responses = tuple(response_iterator)
+    self.assertEqual(exception_ctx.exception.details(), expected_error_details)
+    self.assertEqual(exception_ctx.exception.code(), grpc.StatusCode.INTERNAL)
+    self.assertEqual(response_iterator.details(), expected_error_details)
+    self.assertEqual(response_iterator.code(), grpc.StatusCode.INTERNAL)
+
+  def testStreamRequestBlockingUnaryResponse(self):
+    requests = tuple(b'\x07\x08' for _ in range(test_constants.STREAM_LENGTH))
+    request_iterator = iter(requests)
+    metadata = (('InVaLiD', 'StreamRequestBlockingUnaryResponse'),)
+    expected_error_details = "metadata was invalid: %s" % metadata
+    with self.assertRaises(ValueError) as exception_ctx:
+      response = self._stream_unary(request_iterator, metadata=metadata)
+    self.assertEqual(expected_error_details, exception_ctx.exception.message)
+
+  def testStreamRequestBlockingUnaryResponseWithCall(self):
+    requests = tuple(b'\x07\x08' for _ in range(test_constants.STREAM_LENGTH))
+    request_iterator = iter(requests)
+    metadata = (('InVaLiD', 'StreamRequestBlockingUnaryResponseWithCall'),)
+    expected_error_details = "metadata was invalid: %s" % metadata
+    multi_callable = _stream_unary_multi_callable(self._channel)
+    with self.assertRaises(ValueError) as exception_ctx:
+      response, call = multi_callable.with_call(
+          request_iterator, metadata=metadata)
+    self.assertEqual(expected_error_details, exception_ctx.exception.message)
+
+  def testStreamRequestFutureUnaryResponse(self):
+    requests = tuple(b'\x07\x08' for _ in range(test_constants.STREAM_LENGTH))
```

Comment:
Why not just
```py
request_iterator = (b'\x07\x08' for _ in range(test_constants.STREAM_LENGTH))
```
rather than these two lines? You should only need to create the sequence value if you need to iterate over its elements more than once.